src/lib.rs

```diff
@@ -2619,7 +2657,7 @@ impl<'a, L: 'static + Ledger, Backend: 'a + WalletBackend<'a, L> + Send + Sync>
         } else {
             let (sender, receiver) = oneshot::channel();
 
-            if state.user_keys.contains_key(&receipt.submitter) {
+            if state.user_keys.contains_key(&receipt.submitters[0]) {
```

Comment:
Above (in `transaction_status`) we consider a transaction "submitted by this wallet" if _all_ of the `submitters` are in `user_keys`. I think that is correct, since the wallet only generates transactions where all of the submitters are in the wallet, and we should try to be consistent here.

```suggestion
            if receipt.submitters.iter().all(|key| state.user_keys.contains_key(key)) {
```